lib/travis/env_vars.rb

```diff
@@ -0,0 +1,85 @@
+# this is going to be extracted to a separate gem
+
+require 'strscan'
+require 'forwardable'
+
+module Travis
+  module EnvVars
+    class String
+      ParseError = Class.new(ArgumentError)
+
+      KEY   = /[^\s=]+/
+      WORD  = /(\\.*|[^'"\s])+/
+      QUOTE = /(['"`]{1})/
+      SPACE = /\s+/
+      EQUAL = /=/
+      SUBSH = /\$\(.*?\)/
```

Comment:
👍 The world is always a little brighter when the regexes are broken up and explained 😄

Reply:
❤️